### ISSUE TITLE
False positive for no-unnecessary-semicolons on empty loops

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [Microsoft/tslint-microsoft-contrib#307]
Both of the following give `"unnecessary semi-colon"` errors with `"no-unnecessary-semicolons": true`

```typescript
for (var i = 0; i < 7; i += 1);
```
```typescript
while (false);
```
